teeio-validator/library/cxl_ide_test_lib/cxl_ide_common.c

```diff
@@ -0,0 +1,229 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 Intel. All rights reserved.
+ *  License: BSD 3-Clause License.
+ **/
+
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+#include "teeio_debug.h"
+#include "hal/base.h"
+#include "ide_test.h"
+#include "cxl_ide_test_common.h"
+
+// CXL-IDE supported config items
+const char* m_cxl_ide_test_configuration_name[] = {
+  "default",
+  "pcrc",
+  "ide_stop",
+  "skid",
+  "containment",
+  NULL
+};
+
+uint32_t m_cxl_ide_config_bitmask = (uint32_t)CXL_LINK_IDE_CONFIGURATION_BITMASK;
+
+ide_test_config_funcs_t m_cxl_ide_config_funcs[CXL_IDE_CONFIGURATION_TYPE_NUM] = {
+  {
+    // Default Config
+    cxl_ide_test_config_default_enable,
+    cxl_ide_test_config_default_disable,
+    cxl_ide_test_config_default_support,
+    cxl_ide_test_config_default_check
+  },
+  {NULL, NULL, NULL, NULL},
+  {NULL, NULL, NULL, NULL},
+  {
+    // skid mode
+    cxl_ide_test_config_skid_enable,
+    cxl_ide_test_config_skid_disable,
+    cxl_ide_test_config_skid_support,
+    cxl_ide_test_config_skid_check
+  },
+  {
+    // containment mode
+    cxl_ide_test_config_containment_enable,
+    cxl_ide_test_config_containment_disable,
+    cxl_ide_test_config_containment_support,
+    cxl_ide_test_config_containment_check
+  }
+};
+
+ide_test_group_funcs_t m_cxl_ide_group_funcs = {
+  cxl_ide_test_group_setup,
+  cxl_ide_test_group_teardown
+};
+
+#define CXL_IDE_TEST_CLASS_CASE_NAMES "IdeStream"
+ide_test_case_name_t m_cxl_ide_test_case_names[] = {
+  {"Query",       "1,2",                        CXL_MEM_IDE_TEST_CASE_QUERY     },
+  {"KeyProg",     "1,2,3,4,5,6,7,8,9",          CXL_MEM_IDE_TEST_CASE_KEYPROG   },
+  {"KSetGo",      "1",                          CXL_MEM_IDE_TEST_CASE_KSETGO    },
+  {"KSetStop",    "1",                          CXL_MEM_IDE_TEST_CASE_KSETSTOP  },
+  {"GetKey",      "1",                          CXL_MEM_IDE_TEST_CASE_GETKEY    },
+  {"Test",        CXL_IDE_TEST_CLASS_CASE_NAMES,CXL_MEM_IDE_TEST_CASE_TEST      },
+  {NULL,          NULL,                         CXL_MEM_IDE_TEST_CASE_NUM       }
+};
+
+
+ide_test_case_funcs_t m_cxl_ide_test_case_funcs[CXL_MEM_IDE_TEST_CASE_NUM][MAX_CXL_CASE_ID] = {
+  // Query
+  {
+    {NULL, NULL, NULL, false},
```

Comment:
what's this?

Reply:
Test cases for Query/KeyProg/KSetGo/KSetStop/GetKey have not been implemented. These are the placeholders in m_cxl_ide_test_case_funcs.
In the next version, above funcs will be added with dumb functions like below:
```
bool cxl_ide_test_query_1_setup(void *test_context)
{
  NOT_IMPLEMENTED(__func__);
  return false;
}

bool cxl_ide_test_query_1_run(void *test_context)
{
  NOT_IMPLEMENTED(__func__);
  return false;
}

bool cxl_ide_test_query_1_teardown(void *test_context)
{
  NOT_IMPLEMENTED(__func__);
  return false;
}
```